src/stats/pairhmm/homopolypairhmm.rs

```diff
@@ -290,30 +294,30 @@ impl HomopolyPairHMM {
                     + LogProb::ln_sum_exp(
                         &MATCH_STATES
                             .iter()
-                            .map(|&s| transition_probs[s >> GapY] + v[prev][s][j_])
-                            .chain(once(transition_probs[GapY >> GapY] + v[prev][GapY][j_]))
+                            .map(|&s| transition_probs[&(s >> GapY)] + v[prev][s][j_])
+                            .chain(once(transition_probs[&(GapY >> GapY)] + v[prev][GapY][j_]))
```

Comment:
Is this a syntax equivalent to `transition_probs.get(&(s >>GapY))`? Just wondering, why this is not the same change as above.

Reply:
No, it isn't equivalent: indexing via `[]` will panic if the key doesn't exist, while `get` will return an Option. However, in all but one case we know exactly that those keys actually appear (i.e. the transition probabilities for those specific transitions have been pre-computed), leaving us with one case where there are impossible transitions, hence using `unwrap_or(LogProb::zero())` to provide a default.

Reply:
i.e. we could replace `STATES` with something more specific which avoids transitions from `s` to `m` (`s >> m`) that have an associated probability of zero.